### DISCUSSION TITLE
Return the matched line and nothing else?

### DISCUSSION THREAD

**Discussion by chadbaldwin** [BurntSushi/ripgrep#2427] [Q&A] [answered]
I already have a set of switches to get the output I'm hoping for, but I'm curious if maybe there's a more succient way to handle this?

Context: I have a directory with a bunch of CSV files, I want to extract certain lines and consolidate those into a new CSV file.

I realized I could use ripgrep for this and it seems to be MUCH faster than PowerShell...however, I need the output to purely be the matched line...no extra whitespace (preferably), no line numbers, file names, headers, etc.

I've scanned through the help output at least a few times now but haven't seen any shortcut way to do this.

This is what I've got so far:

```powershell
rg -IN --color never --no-heading 'foo,bar,baz'
```

Here I'm using these switches:

* `-I` - no filename
* `-N` - no line number
* `--color never` - no coloring - probably not necessary for what I'm doing, the coloring will just get removed when I pipe out to a file anyway
* `--no-heading` - don't group matches by each file - removes empty lines between results

So this gets me exactly what I want, I was just curious to see if there was maybe a switch I didn't see that might get what I want.

Obviously I don't _need_ this since I already have a working solution. I'm just posting here out of curiosity.

**Comment by chadbaldwin** [comment, marked as answer]
Ah! I just realized that if I pipe the output to another command in PowerShell, that most of the extra "junk" gets stripped out anyway 🤦‍♂️

```powershell
rg -I 'foo,bar,baz' | Write-Host
```

So that gets me exactly what I wanted.

**Reply by BurntSushi** [reply]
That's pretty much right, yes. When ripgrep is printing to a tty, it will: 1) try to colorize, 2) use a "heading" format and 3) show line numbers. File names are always part of the output unless you're only searching one file or if you use the `-I/--no-filename` flag.

Just relying on ripgrep to squash most of that stuff when piping to a non-tty is totally fine. That is guaranteed behavior and that will never change. It is meant to "degrade" to grep's format in precisely those cases to make this sort of task work how you'd expect.
